pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -668,6 +668,16 @@ SocketAddress serverAddrees() {
                 future.completeExceptionally(writeFuture.cause());
             }
         });
+        eventLoopGroup.schedule(() -> {
+            CompletableFuture<ProducerResponse> requestFuture = pendingRequests.remove(requestId);
+            if (requestFuture != null && !requestFuture.isDone()) {
+                log.warn("{} request {} timed out after {} ms", ctx.channel(), requestId, operationTimeoutMs);
+                future.completeExceptionally(
```

Comment:
We should probably only print the warning if we timed it out. eg: 

```java
if (requestFuture != null 
    && !requestFuture.isDone()
    && future.completeExceptionally(
                new TimeoutException(requestId + 
                      " lookup request timedout after ms " + operationTimeoutMs)
   ) { 
   log.warn("....");
}
```

Also, shouldn't `requestFuture` be marked as failed too?

Reply:
actually `requestFuture` and `future` both are same. I just correct by using `requestFuture` only.